packages/frontend/src/utils/data-decoding.ts

```diff
@@ -50,9 +62,9 @@ export const decodeKPITokenData = async (
     return collaterals.some((collateral) => !collateral)
         ? null
         : {
-              collaterals: collaterals as CollateralData[],
+              collaterals: collaterals.slice() as CollateralData[],
```

Comment:
Why do we need `slice` here? (below too)

Reply:
We don't actually need this, the cast is enough. The problem is that `viem` returns readonly stuff from `decodeAbiParameters` and the slice was needed to make it writeable basically. Addressed in a392c00c4a1c3caaca5352180d9fedb930fbcc87